### PR TITLE
Fix for "Send messages sent to "Logger.fatal" to email or other more visible means"

### PR DESCRIPTION
**Pull request by prinzdezibel**
Resolves #818   
Type: **feature**

## Issue
We want to notify sysadmin if we hit a fatal error so that they can get their site back up.

## Note
**This PR has a companion PR in repository @reactioncommerce/reaction**

https://github.com/reactioncommerce/reaction/pull/4179

## Solution / Changes

Changes:
  - Introduce a new emailstream logger.
  - Register this new logger, if env variable
  REACTION_ENABLE_ALERT_NOTIFICATIONS is defined
  - export Logger instance as mutable export
  - Subscribe to Reaction hook "afterCoreInit" to setup the new
    email stream logger with SMTP information needed for nodemailer

## Breaking changes
None expected

## Testing
1. Make sure you've the code from the companion PR in @reactioncommerce/logger in place.
This can be tricky if you are intend to incorporate the logger NPM package's code as linked NPM package (as opposed to npm install a published version of it). This is because @reactioncommerce/logger now contains a **peer dependency** (@reactioncommerce/hooks), which is known to be a bit difficult to deal with. See also: http://codetunnel.io/you-can-finally-npm-link-packages-that-contain-peer-dependencies/

TLDR: To test it, you may just copy over the source code from @reactioncommerce/logger to the main project's node_module folder, e.g.
```
cd ~/github/reactioncommerce/reaction/node_modules
cp -Rav ~/github/reactioncommerce/logger .
```

2. Add somewhere in server side code, e.g. in  "cart/addToCart" meteor method
`Logger.fatal("email notification test");`

3. Run reaction with proper SMTP settings, e.g. reaction rc-config
```
export REACTION_ENABLE_ALERT_NOTFICATIONS=1 && export REACTION_ALERT_NOTFICATIONS_EMAIL="<youremail>@reactioncommerce.com" && reaction --settings settings/settings.json
```
4. Navigate to the example product and put it to cart
5. See the fatal error message in the server console
6. Go to your email inbox and see an incoming email containing that error

### CODE REVIEW

**Comment by ticean** [comment]
:wave: Thanks for the submission and work here. Apologies that it took so long to get a review.

Log driven alerts can be easily handled outside of the application scope, so I'd prefer that we don't merge any more custom functionality into the logging system. PR's to improve the data that's logged, add timing, additional context, etc. will be 💖.

There are many cloud based log aggregation services out there that can send an alert based on the content of a log message. Most have free tiers for light usage. [Logentries](https://logentries.com/product/alerting-and-reporting/) has some of the best alerting capabilities in the space. [Loggly](https://www.loggly.com/docs/adding-alerts/), [Papertrail](https://help.papertrailapp.com/kb/how-it-works/alerts/), [AWS CloudWatch](https://aws.amazon.com/cloudwatch/details/) and others are capable as well.

And while it's generally better practice to aggregate logs to a centralized location, you could also handle this directly on the server with a process manager or some other log analyzer.
